### PR TITLE
Updated file upload to an explicit two-step process

### PR DESCRIPTION
**Pull request by peter-kwan**
### Summary
The backend API has been updated to side-step the lambda function content size limit. Updating the CLI to match.

**Don't publish before this backend commit is deployed: https://github.com/praetorian-inc/chaos/commit/f0462411f35d9e61dbbc09f67975521bd8f193a6**

### CODE REVIEW

**Comment by peter-kwan** [comment]
@Ayushi-12  -- I also use this opportunity to update the help text for file and definition uploads.

Also, I understand that you will work on major change in your day time tomorrow. So, feel free to merge this before you start your changes.

File: praetorian_cli/sdk/chariot.py [hunk 1]
```diff
@@ -85,18 +85,18 @@ def unlink(self, username: str, id: str = ""):
...
+            self._upload(path, clss, content)
 
     @verify_credentials
     def _upload(self, name: str, clss: str, content: str):
-        resp = requests.put(f"{self.keychain.api}/file", params={"name": name, "class": clss}, data=content,
-                            allow_redirects=True, headers=self.keychain.headers)
-        process_failure(resp)
+        # It is a two-step upload. The PUT request to the /file endpoint is to get a presigned URL for S3.
+        # There is no data transfer.
+        presigned_url = requests.put(f"{self.keychain.api}/file", params={"name": name, "class": clss},
```
Comment (noah-tutt-praetorian) [inline praetorian_cli/sdk/chariot.py:96]: can probably be in a separate change, but `class` isn't used anymore

File: praetorian_cli/handlers/add.py [hunk 2]
```diff
@@ -26,22 +26,40 @@ def asset(controller, name, dns, priority):
...
-    """ Upload a file """
-    controller.upload(name, "manual")
+def upload(controller, path, name):
+    """
+    Upload a file
+
+    PATH is the path to the file
+
+    --name is an optional parameter for the name of the file in Chariot. You can
+    use forward slashes in the file name.
```
Comment (Ayushi-12) [inline praetorian_cli/handlers/add.py:39]:
```suggestion
    """
    Upload a file

    PATH : File path in the local system
```

File: praetorian_cli/handlers/add.py [hunk 3]
```diff
@@ -26,22 +26,40 @@ def asset(controller, name, dns, priority):
...
 @click.argument('path')
-@click.option('-name', '--name', required=False, help='The risk name definition. Default: the filename used')
+@click.option('-name', '--name', help='The risk name definition. Default: the filename used')
 @cli_handler
 def definition(controller, path, name):
-    """ Upload a definition to use for a risk """
+    """
+    Upload a definition to use for a risk
+
+    PATH is the path to the file containing the definition in markdown format
```
Comment (Ayushi-12) [inline praetorian_cli/handlers/add.py:55]:
```suggestion
    PATH:  File path in the local system
```

File: praetorian_cli/handlers/add.py [hunk 4]
```diff
@@ -26,22 +26,40 @@ def asset(controller, name, dns, priority):
...
 
 @add.command('definition')
 @click.argument('path')
-@click.option('-name', '--name', required=False, help='The risk name definition. Default: the filename used')
+@click.option('-name', '--name', help='The risk name definition. Default: the filename used')
 @cli_handler
 def definition(controller, path, name):
-    """ Upload a definition to use for a risk """
+    """
+    Upload a definition to use for a risk
```
Comment (Ayushi-12) [inline praetorian_cli/handlers/add.py:53]:
```suggestion
    Upload a risk definition in markdown format
```

**Review by Ayushi-12 — APPROVED** [review]

**Pull request merged**